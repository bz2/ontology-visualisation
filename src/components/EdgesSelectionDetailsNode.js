import { connect } from 'redux-zero/react'
import PropTypes from 'prop-types'
import { useTranslation } from 'react-i18next'
import { useState } from 'react'

import { Button } from 'primereact/button'
import actions from '../store/actions'
import { PROPERTIES_WITH_I18N, RESERVED_PROPERTIES } from '../constants/graph'
import { dashedToCapitalisedString } from '../constants/functions'

const EdgesSelectionDetailsNode = ({
  node,
}) => {
  const { t } = useTranslation()
  const [isExpanded, toggleExpanded] = useState(false)

  const nodeProperties = Object.keys(node).filter((key) => !RESERVED_PROPERTIES.includes(key)).sort()

  const {
    label
  } = node

  return (
    <div className="elements-selection-details-node">
      <div className="elements-selection-details-node-button">
        <Button
          label={label}
          aria-label={label}
          onClick={() => toggleExpanded(!isExpanded)}
          iconPos="right"
          icon={isExpanded ? 'pi pi-arrow-down' : 'pi pi-arrow-right'}
        />
      </div>

      {
        isExpanded && (
          <div className="elements-selection-details-table elements-selection-details-table-properties">
            {nodeProperties.map((nodeProperty) => {
              const property = PROPERTIES_WITH_I18N.includes(nodeProperty) ? t(nodeProperty) : dashedToCapitalisedString(nodeProperty)

              return (
                <div
                  key={`details-node-${label}-${nodeProperty}`}
                  className="elements-selection-details-table-row"
                >
                  <div className="light-bold">
                    {property}
                  </div>
                  <div>
                    {node[nodeProperty] || t('null')}
                  </div>
                </div>
              )
            })}
          </div>
        )
      }
    </div>
  )
}

EdgesSelectionDetailsNode.propTypes = {
  node: PropTypes.shape().isRequired,
}

export default connect(
  null,
  actions
)(EdgesSelectionDetailsNode)
