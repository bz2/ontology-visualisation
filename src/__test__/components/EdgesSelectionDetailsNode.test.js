import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import EdgesSelectionDetailsNode from '../../components/EdgesSelectionDetailsNode'

const setup = () => {
  const props = {
    node: {
      label: 'Road',
      rdfsLabel: 'Road',
      rdfAbout: 'http://test.com/Road'
    }
  }

  const component = shallow(<EdgesSelectionDetailsNode {...props} />)

  return {
    component,
    props
  }
}

describe('EdgesSelectionDetailsNode', () => {
  afterEach(() => {
    jest.clearAllMocks()
  })

  it('should match snapshot ', () => {
    const {
      component
    } = setup()

    expect(toJson(component)).toMatchSnapshot()
  })
})
