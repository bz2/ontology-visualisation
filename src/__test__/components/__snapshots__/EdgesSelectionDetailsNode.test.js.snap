// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EdgesSelectionDetailsNode should match snapshot  1`] = `
<Connect
  actions={
    Object {
      "updateStoreValue": [Function],
    }
  }
  mapToProps={null}
  node={
    Object {
      "label": "Road",
      "rdfAbout": "http://test.com/Road",
      "rdfsLabel": "Road",
    }
  }
>
  <Component />
</Connect>
`;
